src/framework/theme/components/chat/chat-message.component.ts

```diff
@@ -81,12 +77,24 @@ import { NbChatMessageFile } from './chat-message-file.component';
                               [message]="message" [latitude]="latitude" [longitude]="longitude">
         </nb-chat-message-map>
 
+        <nb-chat-message-text *ngSwitchCase="'text'"
+                [sender]="sender" [date]="date" [dateFormat]="dateFormat"
+                [message]="message">
+        </nb-chat-message-text>
+
         <nb-chat-message-text *ngSwitchDefault
-                              [sender]="sender" [date]="date" [dateFormat]="dateFormat"
-                              [message]="message">
+                                [sender]="sender" [date]="date" [dateFormat]="dateFormat"
+                                [message]="message">
         </nb-chat-message-text>
       </ng-container>
     </div>
+
+    <ng-template #customTemplate>
+        <nb-chat-message-text
+          [sender]="sender" [date]="date" [dateFormat]="dateFormat" [message]="message">
+        </nb-chat-message-text>
+      <ng-container [ngTemplateOutlet]="getActualTemplate(type)" [ngTemplateOutletContext]="{$implicit: customMessageData}"></ng-container>
```

Comment:
Let's rename `getActualTemplate` to `getTemplateForType` or something like this. And let's also extract context object creation in a method. Both methods are meant to be used internally, so let's also prefix them with an underscore (`_getTemplateForType` and `_getTemplateContext`).